src/monty/io.py

```diff
@@ -54,7 +55,52 @@ def zopen(filename: Union[str, Path], *args, **kwargs) -> IO:
     return open(filename, *args, **kwargs)
 
 
-def reverse_readfile(filename: Union[str, Path]) -> Generator[str, str, None]:
+def _get_line_ending(
```

Comment:
could also make `_get_line_ending` a public function, in which case maybe it belongs in the `monty.os` module.

Reply:
I tend not to encourage users to guess line ending through this function (and make it more like a backup plan), because it seems trying to get the line ending from a file is quite prone to all kinds of unexpected errors.